File: infra/src/main/java/com/beat/infra/notification/slack/vo/text/MarkdownText.java
package com.beat.infra.notification.slack.vo.text;

import static com.beat.infra.notification.slack.vo.SlackConstant.TEXT_TYPE_MARKDOWN;

public record MarkdownText(
	String type,
	String text
) implements Text {

	public static MarkdownText newInstance(String text) {
		return new MarkdownText(TEXT_TYPE_MARKDOWN, text);
	}
}

File: infra/src/main/kotlin/com/beat/infra/external/notification/slack/vo/text/MarkdownText.kt
package com.beat.infra.external.notification.slack.vo.text

import com.beat.infra.external.notification.slack.vo.SlackConstant.TEXT_TYPE_MARKDOWN

data class MarkdownText(
    override val type: String,
    val text: String,
) : Text {
    companion object {
        fun newInstance(text: String): MarkdownText = MarkdownText(TEXT_TYPE_MARKDOWN, text)
    }
}
